ci/azure-patch-crates.yml

```diff
@@ -0,0 +1,16 @@
+steps:
+  - script: |
+      set -e
+
+      # Remove any existing patch statements
+      mv Cargo.toml Cargo.toml.bck
+      sed -n '/\[patch.crates-io\]/q;p' Cargo.toml.bck > Cargo.toml
```

Comment:
Handle the case when the primary `Cargo.toml` file has a patch section already. This is now extracted into a dedicated file in order to reuse from multiple spots.